### PR TITLE
cleanup: remove h2UpgradeMap

### PR DESCRIPTION
**Pull request by hzxuzhonghu**
Signed-off-by: Zhonghu Xu <xuzhonghu@huawei.com>

**Please provide a description of this PR:**

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [x] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

File: pilot/pkg/networking/core/v1alpha3/cluster_builder.go [hunk 1]
```diff
@@ -687,13 +680,7 @@ func (cb *ClusterBuilder) shouldH2Upgrade(clusterName string, direction model.Tr
...
 	}
 
-	if !h2UpgradeMap[upgradeTuple{mesh.H2UpgradePolicy, override}] {
-		log.Debugf("Not upgrading cluster: %v (%v %v)", clusterName, mesh.H2UpgradePolicy, override)
-		return false
-	}
-
-	log.Debugf("Upgrading cluster: %v (%v %v)", clusterName, mesh.H2UpgradePolicy, override)
-	return true
+	return mesh.H2UpgradePolicy == meshconfig.MeshConfig_UPGRADE
```
Comment (ramaraochavali) [inline pilot/pkg/networking/core/v1alpha3/cluster_builder.go:683]: Can you add some comments here and retain the debug log line which says upgrading the cluster as per default mesh config policy?

Comment (hzxuzhonghu) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:683]: I think the log is redundant here,  this is mesh global H2UpgradePolicy.

Comment (ramaraochavali) [reply pilot/pkg/networking/core/v1alpha3/cluster_builder.go:683]: IC

**Review by ramaraochavali — APPROVED** [review]

**Review by howardjohn — COMMENTED** [review]
Can you explain why this was removed? Is there no user facing change, just refactor?

**Comment by hzxuzhonghu** [comment]
Yes, just a minor refactor.

**Pull request merged**